## Recent International Matches Results

In this code, I will extract all informations needed for my World Cup 2022 analysis from *[international_matches.csv](https://github.com/bagasadiwaskita/wc22-analysis/blob/047e1688b8f18fb91f586900f36a219f7e1d7f30/Pre-processing/international_matches.csv)* table. The informations I need to extract in this code is ***team's number of wins, draws, loses, goals scored, goals conceded, goals scored per match, goals conceded per match, winrate, current win streak, longest win streak, current unbeaten streak, longest unbeaten streak, current winless streak, longest winless streak, current lose streak, and longest lose streak.***

### A. Data Preparation

In [1]:
# importing needed libraries
import pandas as pd

In [2]:
# read international_matches.csv
intl_matches = pd.read_csv('international_matches.csv')

# show first 5 rows of intl_matches dataframe
intl_matches.head(5)

ID Tournament        Date Home Team  Home Goals  Away Goals Away Team  \
0   1   Friendly  1872-11-30  Scotland           0           0   England   
1   2   Friendly  1873-03-08   England           4           2  Scotland   
2   3   Friendly  1874-03-07  Scotland           2           1   England   
3   4   Friendly  1875-03-06   England           2           2  Scotland   
4   5   Friendly  1876-03-04  Scotland           3           0   England   

  Win Conditions  Home Stadium  
0            NaN          True  
1            NaN          True  
2            NaN          True  
3            NaN          True  
4            NaN          True

In [3]:
# check dataframe columns datatype
intl_matches.dtypes

ID                 int64
Tournament        object
Date              object
Home Team         object
Home Goals         int64
Away Goals         int64
Away Team         object
Win Conditions    object
Home Stadium        bool
dtype: object

In [4]:
# change 'Date' column's data type
intl_matches['Date'] = intl_matches['Date'].astype('datetime64[ns]')

# check the data types of every column
intl_matches.dtypes

ID                         int64
Tournament                object
Date              datetime64[ns]
Home Team                 object
Home Goals                 int64
Away Goals                 int64
Away Team                 object
Win Conditions            object
Home Stadium                bool
dtype: object

In [5]:
# Check the first 5 rows sorted by Date
intl_matches.sort_values(by=['Date']).head(5)

ID Tournament       Date Home Team  Home Goals  Away Goals Away Team  \
0   1   Friendly 1872-11-30  Scotland           0           0   England   
1   2   Friendly 1873-03-08   England           4           2  Scotland   
2   3   Friendly 1874-03-07  Scotland           2           1   England   
3   4   Friendly 1875-03-06   England           2           2  Scotland   
4   5   Friendly 1876-03-04  Scotland           3           0   England   

  Win Conditions  Home Stadium  
0            NaN          True  
1            NaN          True  
2            NaN          True  
3            NaN          True  
4            NaN          True

In [6]:
# check the info of intl_matches dataframe
intl_matches.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17769 entries, 0 to 17768
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   ID              17769 non-null  int64         
 1   Tournament      17769 non-null  object        
 2   Date            17769 non-null  datetime64[ns]
 3   Home Team       17769 non-null  object        
 4   Home Goals      17769 non-null  int64         
 5   Away Goals      17769 non-null  int64         
 6   Away Team       17769 non-null  object        
 7   Win Conditions  201 non-null    object        
 8   Home Stadium    17769 non-null  bool          
dtypes: bool(1), datetime64[ns](1), int64(3), object(4)
memory usage: 1.1+ MB


There are 2 interesting points taken from the codes above.

- There are 17769 entries on the dataframe, with the first row is on 30 November 1872.
- There are 201 non-null entries from 17769 entries in "Win Conditions" column.

### B. Filter Matches by Date

In this section, I will filter the matches by the date it played. Since this is used for World Cup 2022 analysis, I will filter the matches starts from the last matches of the last world cup, World Cup 2018, ends. Data about world cup matches stored in *[world_cup_matches.csv](https://github.com/bagasadiwaskita/wc22-analysis/blob/047e1688b8f18fb91f586900f36a219f7e1d7f30/Pre-processing/world_cup_matches.csv)*.

In [7]:
# read the data about world cup matches
wc_matches = pd.read_csv('world_cup_matches.csv')

# show last 5 rows of wc_matches dataframe
wc_matches.tail(5)

ID  Year        Date           Stage Home Team  Home Goals  Away Goals  \
895  896  2018  2018-07-07  Quarter-finals    Russia           2           2   
896  897  2018  2018-07-10     Semi-finals    France           1           0   
897  898  2018  2018-07-11     Semi-finals   Croatia           2           1   
898  899  2018  2018-07-14     Third place   Belgium           2           0   
899  900  2018  2018-07-15           Final    France           4           2   

    Away Team                    Win Conditions  Host Team  
895   Croatia  Croatia win on penalties (3 - 4)       True  
896   Belgium                               NaN      False  
897   England                        Extra time      False  
898   England                               NaN      False  
899   Croatia                               NaN      False

The last world cup matches is on 15 July 2018. That means the used data for this analysis starts from 16 July 2018.

In [8]:
# filter the match data based on date
filtered_intl_matches=intl_matches.loc[intl_matches['Date']>='2018-07-16']

# show the first 5 rows of filtered data frame
filtered_intl_matches.sort_values(by=['Date']).head(5)

ID           Tournament       Date    Home Team  Home Goals  \
16482  16483             Friendly 2018-09-05     Slovakia           3   
16483  16484  UEFA Nations League 2018-09-06      Germany           0   
16484  16485  UEFA Nations League 2018-09-06        Wales           4   
16485  16486             Friendly 2018-09-06  Netherlands           2   
16486  16487             Friendly 2018-09-06     Portugal           1   

       Away Goals            Away Team Win Conditions  Home Stadium  
16482           0              Denmark            NaN          True  
16483           0               France            NaN          True  
16484           1  Republic of Ireland            NaN          True  
16485           1                 Peru            NaN          True  
16486           1              Croatia            NaN          True

In [9]:
# check the info of filtered_intl_matches dataframe
filtered_intl_matches.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1287 entries, 16482 to 17768
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   ID              1287 non-null   int64         
 1   Tournament      1287 non-null   object        
 2   Date            1287 non-null   datetime64[ns]
 3   Home Team       1287 non-null   object        
 4   Home Goals      1287 non-null   int64         
 5   Away Goals      1287 non-null   int64         
 6   Away Team       1287 non-null   object        
 7   Win Conditions  26 non-null     object        
 8   Home Stadium    1287 non-null   bool          
dtypes: bool(1), datetime64[ns](1), int64(3), object(4)
memory usage: 91.7+ KB


There are 1287 entries left after the date filtering.

### C. Win Conditions Column Check

Since there are only 26 non-null entries in "Win Conditions" column, we need to understand what is happening on it.

In [10]:
# take a look on entries with non-null "Win Conditions" value
filtered_intl_matches_with_wincon = filtered_intl_matches.loc[filtered_intl_matches['Win Conditions'].notnull()]

# show the filtered data frame with win condition
filtered_intl_matches_with_wincon.sort_values(by=['Date'])

ID                    Tournament       Date     Home Team  \
16626  16627                 AFC Asian Cup 2019-01-21     Australia   
16712  16713           UEFA Nations League 2019-06-09   Switzerland   
16761  16762                  Copa America 2019-06-27        Brazil   
16766  16767                  Copa America 2019-06-29       Uruguay   
16768  16769                      Gold Cup 2019-06-29        Mexico   
16778  16779        African Cup of Nations 2019-07-05       Morocco   
16784  16785        African Cup of Nations 2019-07-08         Ghana   
17050  17051       UEFA Euro qualification 2020-11-12        Serbia   
17165  17166       CONCACAF Nations League 2021-06-03        Mexico   
17185  17186       CONCACAF Nations League 2021-06-06      Honduras   
17268  17269                     UEFA Euro 2021-06-28        France   
17273  17274                     UEFA Euro 2021-07-02   Switzerland   
17279  17280                  Copa America 2021-07-03       Uruguay   
17283  17284                  Copa America 2021-07-06     Argentina   
17284  17285                     UEFA Euro 2021-07-06         Italy   
17290  17291                     UEFA Euro 2021-07-11       England   
17300  17301                    COSAFA Cup 2021-07-16       Senegal   
17303  17304                    COSAFA Cup 2021-07-18  South Africa   
17527  17528                      Arab Cup 2021-12-11       Morocco   
17530  17531                      Arab Cup 2021-12-18         Qatar   
17585  17586        African Cup of Nations 2022-02-03      Cameroon   
17586  17587        African Cup of Nations 2022-02-05      Cameroon   
17587  17588        African Cup of Nations 2022-02-06       Senegal   
17621  17622  FIFA World Cup qualification 2022-03-29       Senegal   
17710  17711  FIFA World Cup qualification 2022-06-13     Australia   
17722  17723                     Kirin Cup 2022-06-14         Chile   

       Home Goals  Away Goals     Away Team                 Win Conditions  \
16626           0           0    Uzbekistan     Australia win on penalties   
16712           0           0       England       England win on penalties   
16761           0           0      Paraguay        Brazil win on penalties   
16766           0           0          Peru          Peru win on penalties   
16768           1           1    Costa Rica        Mexico win on penalties   
16778           1           1         Benin         Benin win on penalties   
16784           1           1       Tunisia       Tunisia win on penalties   
17050           1           1      Scotland      Scotland win on penalties   
17165           0           0    Costa Rica        Mexico win on penalties   
17185           2           2    Costa Rica      Honduras win on penalties   
17268           3           3   Switzerland   Switzerland win on penalties   
17273           1           1         Spain         Spain win on penalties   
17279           0           0      Colombia      Colombia win on penalties   
17283           1           1      Colombia     Argentina win on penalties   
17284           1           1         Spain         Italy win on penalties   
17290           1           1         Italy         Italy win on penalties   
17300           2           2      Eswatini       Senegal win on penalties   
17303           0           0       Senegal  South Africa win on penalties   
17527           2           2       Algeria       Algeria win on penalties   
17530           0           0         Egypt         Qatar win on penalties   
17585           0           0         Egypt         Egypt win on penalties   
17586           3           3  Burkina Faso      Cameroon win on penalties   
17587           0           0         Egypt       Senegal win on penalties   
17621           1           0         Egypt       Senegal win on penalties   
17710           0           0          Peru     Australia win on penalties   
17722           0           0         Ghana         Ghana win on

From the result above, it is implied that when "Win Conditions" column is not null, that means the winning team is determined by penalties. It can be seen that excluding Senegal versus Egypt in 29 March 2022, the scoreboard of home and away team is equal. Now let's take a deeper look about that Senegal versus Egypt match.

In [11]:
# take a look on Egypt matches as home team
filtered_intl_matches_egypthome=filtered_intl_matches.loc[filtered_intl_matches['Home Team']=='Egypt']

# show the filtered data frame with Egypt as home team
filtered_intl_matches_egypthome.sort_values(by=['Date'])

ID                            Tournament       Date Home Team  \
16573  16574  African Cup of Nations qualification 2018-11-16     Egypt   
17568  17569                African Cup of Nations 2022-01-30     Egypt   
17602  17603          FIFA World Cup qualification 2022-03-25     Egypt   

       Home Goals  Away Goals Away Team Win Conditions  Home Stadium  
16573           3           2   Tunisia            NaN          True  
17568           2           1   Morocco            NaN         False  
17602           1           0   Senegal            NaN          True

It is implied that in Senegal versus Egypt case, the tournament is on home and away matches rule, with the first leg match results that Egypt as the home team won 1-0 against Senegal. But since the second leg is 1-0 for Senegal and Senegal win on penalties, it won't get any special treatment in the analysis.

Furthermore, when "Win Conditions" column is null, that means the match result is decided without penalty round. Therefore, I will fill null values in "Win Conditions" column with "Match result decided without penalty round" to make it easier to code later.

In [12]:
# fill null values
filtered_intl_matches['Win Conditions'].fillna('Match result decided without penalty round', inplace=True)

# view the first 10 rows
filtered_intl_matches.head(10)

C:\Users\bagas\AppData\Local\Temp\ipykernel_2980\4144473952.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_intl_matches['Win Conditions'].fillna('Match result decided without penalty round', inplace=True)


ID           Tournament       Date    Home Team  Home Goals  \
16482  16483             Friendly 2018-09-05     Slovakia           3   
16483  16484  UEFA Nations League 2018-09-06      Germany           0   
16484  16485  UEFA Nations League 2018-09-06        Wales           4   
16485  16486             Friendly 2018-09-06  Netherlands           2   
16486  16487             Friendly 2018-09-06     Portugal           1   
16487  16488  UEFA Nations League 2018-09-07        Italy           1   
16488  16489  UEFA Nations League 2018-09-07    Lithuania           0   
16489  16490             Friendly 2018-09-07    Argentina           3   
16490  16491             Friendly 2018-09-07      Ecuador           2   
16491  16492             Friendly 2018-09-07  South Korea           2   

       Away Goals            Away Team  \
16482           0              Denmark   
16483           0               France   
16484           1  Republic of Ireland   
16485           1                 Peru   
16486           1              Croatia   
16487           1               Poland   
16488           1               Serbia   
16489           0            Guatemala   
16490           0              Jamaica   
16491           0           Costa Rica   

                                   Win Conditions  Home Stadium  
16482  Match result decided without penalty round          True  
16483  Match result decided without penalty round          True  
16484  Match result decided without penalty round          True  
16485  Match result decided without penalty round          True  
16486  Match result decided without penalty round          True  
16487  Match result decided without penalty round          True  
16488  Match result decided without penalty round          True  
16489  Match result decided without penalty round         False  
16490  Match result decided without penalty round         False  
16491  Match result decided without penalty round          True

### D. Collecting Needed Informations

I will store collected data into new table. The column I want for the new table is "Team", "Win", "Draw", "Lose", "Win_Rate_Percentage", "Goals_Scored", "Goals_Conceded", "Goals_Scored_per_Match", "Goals_Conceded_per_Match", "Current_Win_Streak", "Longest_Win_Streak", "Current_Unbeaten_Streak", "Longest_Unbeaten_Streak", "Current_Winless_Streak", "Longest_Winless_Streak", "Current_Lose_Streak", and "Longest_Lose_Streak".

The first step to do is to record all team names. They are stored in *[wc22_groups.csv](https://github.com/bagasadiwaskita/wc22-analysis/blob/047e1688b8f18fb91f586900f36a219f7e1d7f30/Pre-processing/wc22_groups.csv)*.

In [13]:
# read the needed table
groups=pd.read_csv('wc22_groups.csv')

# record all team names
teams = [teams for teams in groups['Team']]

# check teams list
teams

['Qatar',
 'Ecuador',
 'Senegal',
 'Netherlands',
 'England',
 'Iran',
 'United States',
 'Wales',
 'Argentina',
 'Saudi Arabia',
 'Mexico',
 'Poland',
 'France',
 'Australia',
 'Denmark',
 'Tunisia',
 'Spain',
 'Costa Rica',
 'Germany',
 'Japan',
 'Belgium',
 'Canada',
 'Morocco',
 'Croatia',
 'Brazil',
 'Serbia',
 'Switzerland',
 'Cameroon',
 'Portugal',
 'Ghana',
 'Uruguay',
 'South Korea']

Next, I will set up all lists and constants thal will be used for saving the needed value temporarily.

In [14]:
# set up some lists
wins=[] # list of team's number of wins
draws=[] # list of team's number of draws
loses=[] # list of team's number of loses
WRs=[] # list of team's number of win rate
GSs=[] # list of team's number of goals scored
GCs=[] # list of team's number of goals conceded
GSPMs=[] # list of team's number of goals scored per match
GCPMs=[] # list of team's number of goals conceded per match
WSCs=[] # list of team's number of current win streak
WSLs=[] # list of team's number of longest win streak
UBSCs=[] # list of team's number of current unbeaten streak
UBSLs=[] # list of team's number of longest unbeaten streak
WLSCs=[] # list of team's number of current winless streak
WLSLs=[] # list of team's number of longest winless streak
LSCs=[] # list of team's number of current lose streak
LSLs=[] # list of team's number of longest lose streak

# set up some constants
w=0 # constant of team's number of wins
d=0 # constant of team's number of draws
l=0 # constant of team's number of loses
wr=0 # constant of team's number of win rate
gs=0 # constant of team's number of goals scored
gc=0 # constant of team's number of goals conceded
gspm=0 # constant of team's number of goals scored per match
gcpm=0 # constant of team's number of goals conceded per match
wsc=0 # constant of team's number of current win streaks
wsl=0 # constant of team's number of longest win streaks
ubsc=0 # constant of team's number of current unbeaten streaks
ubsl=0 # constant of team's number of longest unbeaten streaks
wlsc=0 # constant of team's number of current winless streaks
wlsl=0 # constant of team's number of longest winless streaks
lsc=0 # constant of team's number of current lose streaks
lsl=0 # constant of team's number of longest lose streaks

After all is done, now it is the time to run the main code and collect all datas needed.

In [15]:
# collecting the data

# loop every team
for team in teams:
    # loop every rows in the table
    for index, row in filtered_intl_matches.iterrows():
        # condition when team is the home team
        if team in row['Home Team']:
            # condition when the home team wins
            if row['Home Goals'] > row['Away Goals']:
                w=w+1
                wsc=wsc+1
                ubsc=ubsc+1
                # condition when the current winless streak surpass the longest winless streak
                if wlsc>wlsl:
                    wlsl=wlsc
                wlsc=0
                # condition when the current lose streak surpass the longest lose streak
                if lsc>lsl:
                    lsl=lsc
                lsc=0
            # condition when the home team loses
            elif row['Home Goals'] < row['Away Goals']:
                l=l+1
                # condition when the current win streak surpass the longest win streak
                if wsc>wsl:
                    wsl=wsc
                wsc=0
                # condition when the current unbeaten streak surpass the longest unbeaten streak
                if ubsc>ubsl:
                    ubsl=ubsc
                ubsc=0
                wlsc=wlsc+1
                lsc=lsc+1
            # condition when the match draws
            else:
                # condition when there is penalty round
                if row['Win Conditions']!='Match result decided without penalty round':
                    # condition when home team win on penalties
                    if team in row['Win Conditions']:
                        w=w+1
                        wsc=wsc+1
                        ubsc=ubsc+1
                        # condition when the current winless streak surpass the longest winless streak
                        if wlsc>wlsl:
                            wlsl=wlsc
                        wlsc=0
                        # condition when the current lose streak surpass the longest lose streak
                        if lsc>lsl:
                            lsl=lsc
                        lsc=0
                    # condition when home team lose on penalties
                    else:
                        l=l+1
                        # condition when the current win streak surpass the longest win streak
                        if wsc>wsl:
                            wsl=wsc
                        wsc=0
                        # condition when the current unbeaten streak surpass the longest unbeaten streak
                        if ubsc>ubsl:
                            ubsl=ubsc
                        ubsc=0
                        wlsc=wlsc+1
                        lsc=lsc+1
                # condition when there is no penalty round
                else:
                    d=d+1
                    # condition when the current win streak surpass the longest win streak
                    if wsc>wsl:
                        wsl=wsc
                    wsc=0
                    ubsc=ubsc+1
                    wlsc=wlsc+1
                    # condition when the current lose streak surpass the longest lose streak
                    if lsc>lsl:
                        lsl=lsc
                    lsc=0
            gs=gs + row['Home Goals']
            gc=gc + row['Away Goals']
        # condition when team is the away team
        elif team in row['Away Team']:
            # condition when the away team wins
            if row['Home Goals'] < row['Away Goals']:
                w=w+1
                wsc=wsc+1
                ubsc=ubsc+1
                # condition when the current winless streak surpass the longest winless streak
                if wlsc>wlsl:
                    wlsl=wlsc
                wlsc=0
                # condition when the current lose streak surpass the longest lose streak
                if lsc>lsl:
                    lsl=lsc
                lsc=0
            # condition when the away team loses
            elif row['Home Goals'] > row['Away Goals']:
                l=l+1
                # condition when the current win streak surpass the longest win streak
                if wsc>wsl:
                    wsl=wsc
                wsc=0
                # condition when the current unbeaten streak surpass the longest unbeaten streak
                if ubsc>ubsl:
                    ubsl=ubsc
                ubsc=0
                wlsc=wlsc+1
                lsc=lsc+1
            # condition when the match draws
            else:
                # condition when there is penalty round
                if row['Win Conditions']!='Match result decided without penalty round':
                    # condition when away team win on penalties
                    if team in row['Win Conditions']:
                        w=w+1
                        wsc=wsc+1
                        ubsc=ubsc+1
                        # condition when the current winless streak surpass the longest winless streak
                        if wlsc>wlsl:
                            wlsl=wlsc
                        wlsc=0
                        # condition when the current lose streak surpass the longest lose streak
                        if lsc>lsl:
                            lsl=lsc
                        lsc=0
                    # condition when away team lose on penalties
                    else:
                        l=l+1
                        # condition when the current win streak surpass the longest win streak
                        if wsc>wsl:
                            wsl=wsc
                        wsc=0
                        # condition when the current unbeaten streak surpass the longest unbeaten streak
                        if ubsc>ubsl:
                            ubsl=ubsc
                        ubsc=0
                        wlsc=wlsc+1
                        lsc=lsc+1
                # condition when there is no penalty round
                else:
                    d=d+1
                    # condition when the current win streak surpass the longest win streak
                    if wsc>wsl:
                        wsl=wsc
                    wsc=0
                    ubsc=ubsc+1
                    wlsc=wlsc+1
                    # condition when the current lose streak surpass the longest lose streak
                    if lsc>lsl:
                        lsl=lsc
                    lsc=0
            gs=gs + row['Away Goals']
            gc=gc + row['Home Goals']
    # condition when team didn't play a single international matches
    if (w+d+l)==0:
        gspm=0
        gcpm=0
        wr=0
    # condition when team plays at least one international matches
    else:
        gspm = round(gs/(w+d+l),4)
        gcpm = round(gc/(w+d+l),4)
        wr = round(w*100/(w+d+l),4)
    # condition when the current win streak surpass the longest win streak
    if wsc>wsl:
        wsl=wsc
    # condition when the current unbeaten streak surpass the longest unbeaten streak
    if ubsc>ubsl:
        ubsl=ubsc
    # condition when the current win streak surpass the longest win streak
    if wlsc>wlsl:
        wlsl=wlsc
    # condition when the current unbeaten streak surpass the longest unbeaten streak
    if lsc>lsl:
        lsl=lsc
    # insert the values into the list
    wins.append(w)
    draws.append(d)
    loses.append(l)
    WRs.append(wr)
    GSs.append(gs)
    GCs.append(gc)
    GSPMs.append(gspm)
    GCPMs.append(gcpm)
    WSCs.append(wsc)
    WSLs.append(wsl)
    UBSCs.append(ubsc)
    UBSLs.append(ubsl)
    WLSCs.append(wlsc)
    WLSLs.append(wlsl)
    LSCs.append(lsc)
    LSLs.append(lsl)
    # reset the constants
    w=0
    d=0
    l=0
    wr=0
    gs=0
    gc=0
    gspm=0
    gcpm=0 
    wsc=0
    wsl=0
    ubsc=0
    ubsl=0
    wlsc=0
    wlsl=0
    lsc=0
    lsl=0
        
# create the new table
intl_matches_summary=pd.DataFrame({
    'Team': teams, 'Win': wins, 'Draw': draws, 'Lose': loses, 'Win_Rate_Percentage': WRs, 'Goals_Scored': GSs,
    'Goals_Conceded': GCs, 'Goals_Scored_per_Match': GSPMs, 'Goals_Conceded_per_Match': GCPMs, 'Current_Win_Streak': WSCs,
    'Longest_Win_Streak': WSLs, 'Current_Unbeaten_Streak': UBSCs, 'Longest_Unbeaten_Streak': UBSLs,
    'Current_Winless_Streak': WLSCs, 'Longest_Winless_Streak': WLSLs, 'Current_Lose_Streak': LSCs,
    'Longest_Lose_Streak': LSLs
})

# show the created table
intl_matches_summary

Team  Win  Draw  Lose  Win_Rate_Percentage  Goals_Scored  \
0           Qatar   35    10    18              55.5556           107   
1         Ecuador   18    16    15              36.7347            64   
2         Senegal   33     7     8              68.7500            72   
3     Netherlands   30    11     7              62.5000           111   
4         England   34     8     9              66.6667           122   
5            Iran   27     6     5              71.0526            85   
6   United States   37    11    15              58.7302           119   
7           Wales   21    12    16              42.8571            54   
8       Argentina   33    12     4              67.3469            93   
9    Saudi Arabia   22    13    12              46.8085            63   
10         Mexico   40    11    15              60.6061           110   
11         Poland   21    12    14              44.6809            82   
12         France   31    11     7              63.2653           101   
13      Australia   22     4     6              68.7500            67   
14        Denmark   31     9    10              62.0000           108   
15        Tunisia   30     8    14              57.6923            74   
16          Spain   31    12     7              62.0000           112   
17     Costa Rica   19    11    23              35.8491            53   
18        Germany   25    15     8              52.0833           119   
19          Japan   35     7    11              66.0377           106   
20        Belgium   35     7     7              71.4286           128   
21         Canada   30     4    10              68.1818           122   
22        Morocco   28     9     8              62.2222            77   
23        Croatia   23    11    14              47.9167            81   
24         Brazil   38     9     3              76.0000           111   
25         Serbia   24    14     8              52.1739            83   
26    Switzerland   23    10    16              46.9388            90   
27       Cameroon   18    12     9              46.1538            44   
28       Portugal   29    12     7              60.4167           104   
29          Ghana   15    11    10              41.6667            40   
30        Uruguay   20    10    14              45.4545            63   
31    South Korea   31    12     6              63.2653            88   

    Goals_Conceded  Goals_Scored_per_Match  Goals_Conceded_per_Match  \
0               69                  1.6984                    1.0952   
1               56                  1.3061                    1.1429   
2               27                  1.5000                    0.5625   
3               44                  2.3125                    0.9167   
4               34                  2.3922                    0.6667   
5               20                  2.2368                    0.5263   
6               55                  1.8889                    0.8730   
7               51                  1.1020                    1.0408   
8               27                  1.8980                    0.5510   
9               36                  1.3404                    0.7660   
10              62                  1.6667                    0.9394   
11              53                  1.7447                    1.1277   
12              40                  2.0612                    0.8163   
13              18                  2.0938                    0.5625   
14              39                  2.1600                    0.7800   
15              38                  1.4231                    0.7308   
16              37                  2.2400                    0.7400   
17              58                  1.0000                    1.0943   
18              57                  2.4792                    1.1875   
19              39                  2.0000                    0.7358   
20              43                  2.6122                    0.8776   
21              31 

Now all the needed international matches data have collected. The last thing to do is to save the data into a file.

In [16]:
# save the collected data into csv
intl_matches_summary.to_csv('intl_matches_summary.csv', index=False)

The collected data is stored in *[intl_matches_summary.csv](https://github.com/bagasadiwaskita/wc22-analysis/blob/047e1688b8f18fb91f586900f36a219f7e1d7f30/Pre-processing/intl_matches_summary.csv)*.